jwst/datamodels/model_base.py

```diff
@@ -400,71 +399,17 @@ def callback(schema, path, combiner, ctx, recurse):
-        Return datatype and dimension for each array or table
-        """
-        def get_field_info(path, instance):
-            field_info = []
-            if isinstance(instance, dict):
-                if path:
-                    path += '.'
-                for name, val in instance.items():
-                    new_path = path + name.lower()
-                    field_info.extend(get_field_info(new_path, val))
-            elif isinstance(instance, list):
-                for index, val in enumerate(instance):
-                    new_path = "%s[%d]" % (path, index)
-                    field_info.extend(get_field_info(new_path, val))
-            elif isinstance(instance, np.ndarray):
-                if instance.shape[0] > 0:
-                    shape_info = get_shape_info(instance)
-                    type_info = get_type_info(instance)
-                    field_info = [(path, shape_info, type_info)]
-            return field_info
-
-        def get_meta_info(meta):
-            meta_info = []
-            for attribute in ('filename', 'date', 'model_type'):
-                if hasattr(meta, attribute):
-                    value = getattr(meta, attribute)
-                    if value is not None:
-                        meta_info.append(attribute + ': ' + value)
-            return meta_info
-
-        def get_shape_info(instance):
-            if hasattr(instance, '_coldefs'):
-                nrows = instance.shape[0]
-                ncols = len(instance._coldefs)
-                shape_info = "{}R x {}C".format(nrows, ncols)
-            else:
-                # Display shape in fits order (reversed)
-                shape = [str(s) for s in reversed(instance.shape)]
-                shape_info = '(' + ','.join(shape) + ')'
-            return shape_info
-
-        def get_type_info(instance):
-            if hasattr(instance, '_coldefs'):
-                col_info = []
-                for coldef in instance._coldefs:
-                    col_info.append(coldef.name + ':' + coldef.format)
-                type_info = '(' + ','.join(col_info) + ')'
-            else:
-                type_info = numpy_dtype_to_asdf_datatype(instance.dtype)[0]
-            return type_info
-
-        meta_info = get_meta_info(self.meta)
-        field_info = get_field_info('', self._instance)
+    def info(self, **kwargs):
+        return self._asdf.info(**kwargs)
 
+    def search(self, *args, **kwargs):
+        return self._asdf.search(*args, **kwargs)
 
-        buffer = meta_info
-        format_string = "%-40s%-20s%s"
-        buffer.append(70 * '-')
-        buffer.append(format_string % ('attribute', 'size', 'type'))
-        buffer.append(70 * '-')
-        for field in field_info:
-            buffer.append(format_string % field)
-        return "\n".join(buffer) + "\n"
+    try:
+        info.__doc__ = AsdfFile().info.__doc__
+        search.__doc__ = AsdfFile().search.__doc__
+    except AttributeError:
+        pass
```

Comment:
I think you can use the class instead of the instance here.  I.e.:

```suggestion
    try:
        info.__doc__ = AsdfFile.info.__doc__
        search.__doc__ = AsdfFile.search.__doc__
    except AttributeError:
        pass
```